### PR TITLE
fix: 修复闭包导致的pagination被错误重置的问题

### LINKED ISSUE #1738: 🐛 分页组件 依旧 无法切换 pageSize, pageCode
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       |    lastest      |
| @antv/s2-react |  lastest        |
| @antv/s2-vue   |          |

### Sheet Type

- [ ] PivotSheet
- [-] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet

### 🖋 Description

https://user-images.githubusercontent.com/46083022/188841370-2558fc63-6df6-4ac3-bac3-484de5360c29.mov

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1738 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
关联pr:https://github.com/antvis/S2/pull/1722
### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

**Review by xingwanying — APPROVED** [review]
LG

File: packages/s2-react/src/hooks/usePagination.ts [hunk 1]
```diff
@@ -62,14 +62,14 @@ export const usePagination = (s2: SpreadSheet, props: SheetComponentsProps) => {
...
+    const totalUpdateCallback = (data) => {
+      setPagination((prev) => ({ ...prev, total: data.total }));
+    };
     s2.on(S2Event.LAYOUT_PAGINATION, totalUpdateCallback);
     return () => {
       s2.off(S2Event.LAYOUT_PAGINATION, totalUpdateCallback);
     };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [s2]);
+  }, [paginationCfg, s2]);
```
Comment (wjgogogo) [inline packages/s2-react/src/hooks/usePagination.ts:72]: 在DI出现了选择页码后，又被重置的问题。但是在playground中却没有体现。关键在于：
在playground中，初始化是未开启分页的，所以在开启分页后，由于s2本身没有变化，所有根本没有将该hook挂在evet上。

**Pull request merged**